chainstate-storage/src/internal/test.rs

```diff
@@ -277,27 +277,32 @@ pub fn create_rand_block_undo(
     rng: &mut impl Rng,
     max_lim_of_utxos: u8,
     max_lim_of_tx_undos: u8,
-    block_height: BlockHeight,
 ) -> BlockUndo {
     let mut counter: u64 = 0;
 
-    let mut block_undo: Vec<TxUndo> = vec![];
+    let mut reward_utxos: Vec<Utxo> = vec![];
+    let utxo_rng = rng.gen_range(1..max_lim_of_utxos);
+    for i in 0..utxo_rng {
+        counter += u64::from(i);
```

Comment:
Is this like a series sum, from 0 to n-1, yielding n(n-1)/2? Why?

Or is this a mistake and you just need a counter, in which case you could just use i?

Unlikely the former, so in case it's the latter, this can be simplified with a generator function pattern: `(0..utxo_rng).into_iter().map(|_|create_rand_utxo(rng, counter))` or something similar.

Reply:
That's just the way the rest of the test was written. I'm not sure about the reasoning, but I agree that it makes sense to simplify.